server/task/src/test/java/org/apache/james/task/MemoryTaskManagerTest.java

```diff
@@ -87,28 +99,22 @@ public void taskCodeAfterCancelIsNotRun() {
             return Task.Result.COMPLETED;
         });
 
+        sleep(TIME_TO_ENQUEUE);
```

Comment:
Does this means that when sublmit returns the task is actually not yet submitted, and I have no way to wait for it to be?

Sounds like a future source of bugs to me!

Reply:
Yes.

It's just the time for reactor to publish the task  on another Schedulers.
If we don't use publishOn(aSchedulers) or use an async Processor like the WorkQueueProcessor,
the task will be published on the current thread and the tests full of locks with countdown latches will deadlock.
I don't think it's an issue in real world usage.